src/main/java/org/jsoup/nodes/Document.java

```diff
@@ -206,6 +209,79 @@ public Element text(String text) {
     public String nodeName() {
         return "#document";
     }
+    
+    /**
+     * Sets the charset used in this document. This method is equivalent
+     * to {@link OutputSettings#charset(java.nio.charset.Charset)
+     * OutputSettings.charset(Charset)} but in addition it updates the
+     * charset / encoding element within the document.
+     * 
+     * <p>This only applies if {@link #updateMetaCharset(boolean)
+     * updateMetaCharset} set to <tt>true</tt>; otherwise there are no elements
+     * changed and the new value is delegated to
+     * {@link OutputSettings#charset(java.nio.charset.Charset) 
+     * OutputSettings.charset(Charset)} only.</p>
+     * 
+     * <p>If there's no element with charset / encoding information yet it will
+     * be created. Obsolete charset / encoding definitions are removed!</p>
+     * 
+     * <p><b>Elements used:</b></p>
+     * 
+     * <ul>
+     * <li><b>Html:</b> <i>&lt;meta charset="CHARSET"&gt;</i></li>
+     * <li><b>Xml:</b> <i>&lt;?xml version="1.0" encoding="CHARSET"&gt;</i></li>
+     * </ul>
+     * 
+     * @param charset Charset
+     * 
+     * @see #updateMetaCharset(boolean) 
+     * @see OutputSettings#charset(java.nio.charset.Charset) 
+     */
+    public void charset(Charset charset) {
```

Comment:
I think that it would be better to turn on updateMetaCharset when this method is hit. If the user didn't want it updated, they wouldn't hit this method. They would just use the Output settings.